File: given_parser/ast/IndexExpression.java
package ast;

import java.util.Hashtable;
import java.util.LinkedHashMap;
import java.util.List;

public class IndexExpression
    extends AbstractExpression
{
    private final Expression left;
    private final Expression index;

    public IndexExpression(int lineNum, Expression left, Expression index)
    {
        super(lineNum);
        this.left = left;
        this.index = index;
    }
}

File: given_parser/myasts/IndexExpression.cpp
#include "IndexExpression.hpp"

namespace ast {

// Constructor
IndexExpression::IndexExpression(int lineNum, std::shared_ptr<Expression> left, std::shared_ptr<Expression> index)
    : AbstractExpression(lineNum), left(left), index(index) {}

// Getter for left expression
std::shared_ptr<Expression> IndexExpression::getLeft() const {
    return left;
}

// Getter for index expression
std::shared_ptr<Expression> IndexExpression::getIndex() const {
    return index;
}

} // namespace ast

